File: examples/ScriptingService/Notifications.java

import omero.api.*;
import omero.model.*;
import omero.grid.*;

public class Notifications {

    public final static String SCRIPT = "" +
    "import omero\n" +
    "import omero.scripts as s\n" +
    "s.client(\"name\")\n";
    // does nothing
    public static void main(String args[]) throws Exception{

        long launched = System.currentTimeMillis();
        omero.client client = new omero.client(args);
        try {
            ServiceFactoryPrx sf = client.createSession();
            IScriptPrx scriptService = sf.getScriptService();
            long id = scriptService.uploadOfficialScript(
                String.format("/examples/%s.py", java.util.UUID.randomUUID()), SCRIPT);
            ScriptProcessPrx proc = scriptService.runScript(id, null, null);
            ProcessCallbackI cb = new ProcessCallbackI(client, proc);
            launched = System.currentTimeMillis();
            while (null == cb.block(500)) {
                if (10000 < (System.currentTimeMillis() - launched)) {
                    throw new RuntimeException("Too long!");
                }
            }
        } finally {
            System.out.println("Finished in (ms): " + (System.currentTimeMillis() - launched));
            client.closeSession();
        }
   }
}

File: examples/ScriptingService/Notifications.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

SCRIPT = """if True:
    import omero
    import omero.scripts as s
    s.client("name")"""

import omero
import omero.scripts
import time
import uuid

launched = time.time()
client = omero.client()
try:
    sf = client.createSession()
    scriptService = sf.getScriptService()
    id = scriptService.uploadOfficialScript(
        "/examples/%s.py" % uuid.uuid4(), SCRIPT)
    proc = scriptService.runScript(id, None, None)
    cb = omero.scripts.ProcessCallbackI(client, proc)
    launched = time.time()
    while cb.block(500) is None:
        if 10.0 < (time.time() - launched):
            raise StopIteration("Too long!")
finally:
    print "Finished in (s): %s" % (time.time() - launched)
    client.closeSession()
